refactor(router): deprecate no longer needed `resolver` fields

AndrewKushnir opened the pull request:
DEPRECATED:

The `resolver` argument of the `RouterOutletContract.activateWith` function and the `resolver` field of the `OutletContext` class are deprecated. Passing component factory resolvers are no longer needed. The `ComponentFactoryResolver`-related symbols were deprecated in `@angular/core` package since v13.

## PR Type
What kind of change does this PR introduce?

- [x] Other... Please describe: deprecation of no longer relevant APIs.


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

dylhunn reviewed (approved):
reviewed-for: public-api

atscott committed:
feat(router): Add `EnvironmentInjector` to `RouterOutlet.activateWith`

The current API for `activateWith` uses the deprecated
`ComponentFactoryResolver`. The replacement for this is to use
`EnvironmentInjector`. That is, instead of

```
const factory = componentFactoryResolver.resolveComponentFactory(component);
location.createComponent(factory, location.length, injector);
```

the replacement would be

```
location.createComponent(
    component, {index: location.length, injector, environmentInjector});
```

atscott committed:
refactor(router): deprecate no longer needed resolver fields

DEPRECATED:

The `resolver` argument of the `RouterOutletContract.activateWith` function and the `resolver` field of the `OutletContext` class are deprecated. Passing component factory resolvers are no longer needed. The `ComponentFactoryResolver`-related symbols were deprecated in `@angular/core` package since v13.

atscott reviewed (approved):
reviewed-for: public-api

AndrewKushnir commented:
[Presubmit (TGP)](https://test.corp.google.com/OCL:442688354:BASE:442688346:1650333658321:85fed457).

jessicajaniuk reviewed (approved):
LGTM :cookie:

reviewed-for: public-api

AndrewKushnir commented:
Note to the Caretaker: this PR would require [a local patch in g3](http://cl/442852042) (that should be added to the sync CL), which would also require extra LGTMs.

dylhunn commented:
This looks like we need Pantheon approvals to submit the g3 CL. Would you be able to send a CL to get those? Then we can submit the CL at the same time that we sync the PR.

AndrewKushnir commented:
> This looks like we need Pantheon approvals to submit the g3 CL. Would you be able to send a CL to get those? Then we can submit the CL at the same time that we sync the PR.

@dylhunn I think we need those changes in the same sync CL, so that we don't get into a broken state (it might be hard to submit at the exact same time) and can rollback all changes all at once if needed.

dylhunn commented:
This PR was merged into the repository by commit 1d2f5c1101ccd182f5b528de52583a1b98dd6789.
